internal/app/cf-terraforming/cmd/generate_test.go

```diff
@@ -148,7 +149,17 @@ func TestResourceGeneration(t *testing.T) {
 			storedAccountID := viper.GetString("account")
 			storedZoneID := viper.GetString("zone")
 
-			if tc.identiferType == "account" {
+			// edge case for the weird workflow that is cloudflare_worker_routes
+			if tc.testdataFilename == "cloudflare_worker_route_routes" {
```

Comment:
this feels like something is really wrong here. we shouldn't be basing this logic on specific test cases (especially here when we have conflicting flags `--zone` and `--account`). we should only have one or the other.

Reply:
Normally i'd agree, but due to the nature of the way Cloudflare-go does the worker api call, its necessary for both to exist so cf-terraforming can init Cloudflare-go with the accountID. 
https://github.com/cloudflare/cloudflare-go/blob/master/workers.go#L674 

Personally, i'd love to just default to `routes` in Cloudflare-go and it fixes all of this.

Reply:
the flags are intended to be mutually exclusive so if this is working, it’s leveraging unintended and incorrect behaviour. there isn’t a case where you want to be using an account _and_ zone together; it’s one or the other. 

I’m happy to see this fixed in the Go library itself but we’ll just need to confirm the behaviour with the Workers team and the downstream libraries (like Terraform).